codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/plugins/DefaultAWSAuthSchemePlugin.kt

```diff
@@ -45,13 +46,21 @@ class DefaultAWSAuthSchemePlugin(
                         "Default${AuthSchemeResolverGenerator.getSdkId(ctx)}AuthSchemeResolver()",
                     )
                     writer.write("config.authSchemes = \$L", AWSAuthUtils(ctx).getModeledAuthSchemesSupportedBySDK(ctx, writer))
-                    writer.write(
-                        "config.awsCredentialIdentityResolver = \$N()",
-                        AWSSDKIdentityTypes.DefaultAWSCredentialIdentityResolverChain,
-                    )
+                    if (ctx.settings.internalClient) {
+                        writer.write(
+                            "config.awsCredentialIdentityResolver = \$N()",
+                            InternalAWSCommonTypes.EmptyAWSCredentialIdentityResolver,
+                        )
+                    } else {
+                        writer.write(
+                            "config.awsCredentialIdentityResolver = \$N()",
+                            AWSSDKIdentityTypes.DefaultAWSCredentialIdentityResolverChain,
+                        )
+                    }
```

Comment:
Above, internal clients get a private type of AWSCredentialIdentityResolver that does not depend on AWSSDKIdentity.  This, combined with breaking a couple other dependency cycles in runtime modules, means that the internal clients have no reference back to AWSSDKIdentity, and the credential resolvers in those packages can create and use those internal clients directly.

External clients continue to get the same "default chain" identity resolver from AWSSDKIdentity.